sdk/scheduler/src/main/java/com/mesosphere/sdk/offer/evaluate/TaskResourceMapper.java

```diff
@@ -230,9 +230,11 @@ private static OfferEvaluationStage toEvaluationStage(
             Optional<String> resourceId,
             Optional<String> persistenceId) {
         if (resourceSpec instanceof NamedVIPSpec) {
-            return new NamedVIPEvaluationStage((NamedVIPSpec) resourceSpec, taskSpecName, resourceId);
+            NamedVIPSpec namedVIPSpec = (NamedVIPSpec) resourceSpec;
```

Comment:
Probably because, like you imply, the logical distance between the two in the codebase is counterintuitively high, but as is in evidence here, evaluation stages directly depend on their attendant specs. We've just elected now to opt out of having the type system help us to manage those dependencies.